router/worker.go

```diff
@@ -553,6 +554,12 @@ func (w *worker) processDestinationJobs() {
 				}
 				errorAt = routerutils.ERROR_AT_TF
 			}
+		} else if destinationJob.StatusCode == utilTypes.FilterEventCode || destinationJob.StatusCode == utilTypes.SuppressEventCode {
+			respStatusCode = destinationJob.StatusCode
+			respBody = destinationJob.Error
+			if respBody == "" {
+				respBody = utilTypes.GetFallBackErrorMessage(respStatusCode)
+			}
```

Comment:
instead of introducing yet another branch, we could "reuse" the existing `else` branch, with the hope of eventually doing something better about the `errorAt` flag, which is used only for stats purposes:

```go
respStatusCode = destinationJob.StatusCode
respBody = destinationJob.Error
switch destinationJob.StatusCode {
case utilTypes.FilterEventCode:
	if respBody == "" {
		respBody = "Event filtered"
	}
case utilTypes.SuppressEventCode:
	if respBody == "" {
		respBody = "Event handled by transformer"
	}
default:
	errorAt = routerutils.ERROR_AT_TF
}
```

Reply:
Cleaned up.